tests/UI/pages/BO/BObasePage.js

```diff
@@ -278,18 +383,112 @@ class BOBasePage extends CommonPage {
    * @returns {Promise<void>}
    */
   async goToSubMenu(page, parentSelector, linkSelector) {
-    if (await this.elementNotVisible(page, `${parentSelector}.open`, 1000)) {
+    await this.clickSubMenu(page, parentSelector);
+    await this.scrollTo(page, linkSelector);
+    await this.clickAndWaitForNavigation(page, linkSelector);
+    if (await this.isSidebarCollapsed(page)) {
+      await this.waitForHiddenSelector(page, `${linkSelector}.link-active`);
+    } else {
+      await this.waitForVisibleSelector(page, `${linkSelector}.link-active`);
+    }
+  }
+
+  /**
+   * Open a subMenu
+   * @param page {Page} Browser tab
+   * @param parentSelector {string} Selector of the parent menu
+   * @returns {Promise<void>}
+   */
+  async clickSubMenu(page, parentSelector) {
+    const openSelector = await this.isSidebarCollapsed(page) ? '.ul-open' : '.open';
+    if (await this.elementNotVisible(page, `${parentSelector}${openSelector}`, 1000)) {
       // open the block
       await this.scrollTo(page, parentSelector);
 
       await Promise.all([
         page.click(parentSelector),
-        this.waitForVisibleSelector(page, `${parentSelector}.open`),
+        this.waitForVisibleSelector(page, `${parentSelector}${openSelector}`),
       ]);
     }
-    await this.scrollTo(page, linkSelector);
-    await this.clickAndWaitForNavigation(page, linkSelector);
-    await this.waitForVisibleSelector(page, `${linkSelector}.link-active`);
+  }
+
+  /**
+   * Return is a submenu is active
+   * @param page {Page} Browser tab
+   * @param linkSelector {string} Selector of the menu
+   * @return {Promise<boolean>}
+   */
+  async isSubMenuActive(page, linkSelector) {
+    return (await page.$$(`${linkSelector}.link-active`)).length > 0;
+  }
+
+  /**
+   * Return is the navbar is visible
+   * @param page {Page} Browser tab
+   * @return {Promise<boolean>}
+   */
+  async isNavbarVisible(page) {
+    return this.elementVisible(page, '.nav-bar:not(.mobile-nav)', 1000);
```

Comment:
Please create a new variable for the selector

Reply:
Fixed